linker/jvm/src/test/scala/org/scalajs/linker/PathOutputDirectoryTest.scala

```diff
@@ -54,4 +58,33 @@ class PathOutputDirectoryTest {
       assertEquals(lastModifiedBefore, lastModifiedAfter)
     }
   }
+
+  @Test
+  def sanePosixFilePermissions(): AsyncResult = await {
+    import java.nio.file.attribute.PosixFilePermission._
+
+    val tmpDir = Files.createTempDirectory("test-dir")
```

Comment:
Consider using Jimfs. This will even allow you to configure whether the file system supports posix attributes and hard assert that it works. And you can write a (platform independent) test making sure that `PathOutputDirectory` does not fail if the underlying FS doesn't support posix attributes.